continuationsPlugin/src/main/scala/continuations/DefDefTransforms.scala

```diff
@@ -60,50 +60,91 @@ object DefDefTransforms extends TreesChecks:
 
   private def transformSuspendContinuationBody(
       callSuspensionPoint: tpd.Tree,
-      safeContinuationRef: tpd.Tree)(using Context): List[tpd.Tree] = {
+      safeContinuationRef: tpd.Tree)(using Context): tpd.Tree = {
     val resumeMethod = safeContinuationRef.select(termName("resume"))
 
-    val suspendContinuationBody: List[tpd.Tree] =
-      callSuspensionPoint
-        .filterSubTrees {
-          case Trees.DefDef(_, _, _, _) => true
-          case _ => false
-        }
-        .flatMap {
-          case Trees.DefDef(_, _, _, Trees.Block(stats, expr)) =>
-            stats.map(s => s.withType(s.typeOpt)) :+ expr.withType(expr.typeOpt)
-          case tree =>
-            tree.asInstanceOf[tpd.DefDef].rhs :: Nil
-        }
-
-    val removedAnonFunc: List[Symbol] = suspendContinuationBody
-      .flatMap(
-        _.filterSubTrees(t =>
-          t.symbol.exists &&
-            t.symbol.owner.isAnonymousFunction &&
+    val anonymousContinuationFunctions = callSuspensionPoint.filterSubTrees {
+      case t @ Trees.DefDef(_, _, _, _)
+          if t.symbol.exists &&
+            t.symbol.isAnonymousFunction
+            &&
             t.symbol
-              .owner
               .paramSymss
               .flatten
-              .exists(_.info.hasClassSymbol(requiredClass(continuationFullName)))))
-      .map(_.symbol.owner)
-      .distinct
+              .exists(_.info.hasClassSymbol(requiredClass(continuationFullName))) =>
```

Comment:
Aux Method introduced last week:  
```suggestion
              .exists(hasContinuationClass) =>
```